app/services/authorizer.rb

```diff
@@ -104,10 +104,18 @@ def build_filtered_scope_components(resource_class, all_filters, options)
     return result if all_filters.any?(&:unlimited?)
 
     search_string = build_scoped_search_condition(all_filters.select(&:limited?))
-    find_options = ScopedSearch::QueryBuilder.build_query(resource_class.scoped_search_definition, search_string, options)
-    result[:where] << find_options[:conditions]
-    result[:includes].push(*find_options[:include])
-    result[:joins].push(*find_options[:joins])
+
+    begin
+      find_options = ScopedSearch::QueryBuilder.build_query(resource_class.scoped_search_definition, search_string, options)
+
+      result[:where] << find_options[:conditions]
+      result[:includes].push(*find_options[:include])
+      result[:joins].push(*find_options[:joins])
+    rescue ScopedSearch::QueryNotSupported => e
+      Foreman::Logging.logger('permissions').error N_("Scoped search query not supported: #{e.message}")
+      result[:where] << (user.admin? ? '1=1' : '1=0')
```

Comment:
Nitpick: if the user is an admin, we don't need to add any scope, and we can leave the conditions empty (unless there is some code that is not expecting empty conditions down the road).
```suggestion
      result[:where] << '1=0' unless user.admin?
```